src/Estimators/StructureFactorEstimator.cpp

```diff
@@ -103,28 +123,54 @@ void StructureFactorEstimator::write(hdf_archive& file)
   hdf_path hdf_name{my_name_};
   file.push(hdf_name);
   // this is call rhok_e_e in the output of the legacy, but that is just wrong it is |rhok_e_e_|^2
-  file.write(sfk_e_e_, "sfk_e_e");
-  file.write(rhok_e_, "rhok_e_");
+  append_sfk_e_e_position_ = file.append(sfk_e_e_, "sfk_e_e", append_sfk_e_e_position_);
+  append_rhok_e_position_  = file.append(rhok_e_, "rhok_e", append_rhok_e_position_);
+  file.pop();
 }
 
 void StructureFactorEstimator::collect(const RefVector<OperatorEstBase>& type_erased_operator_estimators)
 {
   int num_crowds = type_erased_operator_estimators.size();
   for (OperatorEstBase& crowd_oeb : type_erased_operator_estimators)
   {
-    StructureFactorEstimator& crowd_sfe = dynamic_cast<StructureFactorEstimator&>(crowd_oeb);
+    auto& crowd_sfe = dynamic_cast<StructureFactorEstimator&>(crowd_oeb);
     this->sfk_e_e_ += crowd_sfe.sfk_e_e_;
     this->rhok_e_ += crowd_sfe.rhok_e_;
+    walkers_weight_ += crowd_sfe.walkers_weight_;
```

Comment:
I'm confused by the coverage report. Is this collect function covered by test? IIRC, `collect()` is the most important entry to an estimator.

Reply:
@ye-luo  I will take a look at the coverage